### PR TITLE
4.8.0 mvp

### PR DESCRIPTION
**Pull request by Manas-tarento**
New workflow service implementation for taxonomy

### CODE REVIEW

File: src/main/java/org/sunbird/workflow/service/impl/WorkflowServiceImplV2.java [hunk 1]
```diff
@@ -1,42 +1,354 @@
...
+        if (!CollectionUtils.isEmpty(wfRequest.getUpdateFieldValues())) {
+            List<HashMap<String, Object>> validTerms = new ArrayList<>();
+            ArrayList<String> invalidTerms = new ArrayList<>();
+            for (HashMap<String, Object> updatedField : wfRequest.getUpdateFieldValues()) {
+                if (StringUtils.isEmpty((String) updatedField.get(Constants.APPROVAL_STATUS)) || updatedField.get(Constants.APPROVAL_STATUS).equals(Constants.DRAFT)) {
+                    validTerms.add(updatedField);
+                } else {
+                    invalidTerms.add((String) updatedField.get(Constants.IDENTIFIER));
+                }
+            }
```
Comment (karthik-tarento) [inline src/main/java/org/sunbird/workflow/service/impl/WorkflowServiceImplV2.java:88]: This part of code needs to be moved to validateRequest method

File: src/main/java/org/sunbird/workflow/service/impl/WorkflowServiceImplV2.java [hunk 2]
```diff
@@ -1,42 +1,354 @@
...
+            if (CollectionUtils.isNotEmpty(invalidTerms)) {
+                throw new InvalidDataInputException(Constants.TERM_APPROVAL_STATUS_ERROR + ": " + invalidTerms);
```
Comment (karthik-tarento) [inline src/main/java/org/sunbird/workflow/service/impl/WorkflowServiceImplV2.java:90]: Do not throw exception... always update the response object with error code..

File: src/main/java/org/sunbird/workflow/service/impl/WorkflowServiceImplV2.java [hunk 3]
```diff
@@ -1,42 +1,354 @@
...
+    public Response getWfApplication(String userToken, String wfId) {
+        String userId = validateAuthTokenAndFetchUserId(userToken);
+        WfStatusEntityV2 applicationStatus = wfStatusRepoV2.findByUserIdAndWfId(userId, wfId);
+        List<Map<String, Object>> updateFieldValuesList = null;
+        if (StringUtils.isNotEmpty(applicationStatus.getUpdateFieldValues())) {
+            try {
+                updateFieldValuesList = mapper.readValue(applicationStatus.getUpdateFieldValues(), new TypeReference<List<Map<String, Object>>>() {
+                });
+            } catch (Exception ex) {
+                throw new ApplicationException(Constants.UPDATED_FIELD_VALUES_ERROR_MESSAGE, ex);
```
Comment (karthik-tarento) [inline src/main/java/org/sunbird/workflow/service/impl/WorkflowServiceImplV2.java:122]: Do not throw exception - update the response object with error code.

File: src/main/java/org/sunbird/workflow/service/impl/WorkflowServiceImplV2.java [hunk 4]
```diff
@@ -1,42 +1,354 @@
...
+            }
+        }
+        if (ObjectUtils.isEmpty(wfStatus)) {
+            throw new BadRequestException(Constants.WORKFLOW_STATE_CHECK_ERROR);
+        }
+        return wfStatus;
+    }
+
+    private String validateAuthTokenAndFetchUserId(String authUserToken) {
+        return requestInterceptor.fetchUserIdFromAccessToken(authUserToken);
```
Comment (karthik-tarento) [inline src/main/java/org/sunbird/workflow/service/impl/WorkflowServiceImplV2.java:352]: Avoid methods for single line code

**Review by karthik-tarento — CHANGES_REQUESTED** [review]

File: src/main/java/org/sunbird/workflow/service/impl/WorkflowServiceImplV2.java (hunk 2 above)
Comment (Manas-tarento) [reply src/main/java/org/sunbird/workflow/service/impl/WorkflowServiceImplV2.java:90]: added proper response structure like cb-ext

File: src/main/java/org/sunbird/workflow/service/impl/WorkflowServiceImplV2.java (hunk 3 above)
Comment (Manas-tarento) [reply src/main/java/org/sunbird/workflow/service/impl/WorkflowServiceImplV2.java:122]: updated

File: src/main/java/org/sunbird/workflow/service/impl/WorkflowServiceImplV2.java (hunk 4 above)
Comment (Manas-tarento) [reply src/main/java/org/sunbird/workflow/service/impl/WorkflowServiceImplV2.java:352]: updated

File: src/main/java/org/sunbird/workflow/service/impl/WorkflowServiceImplV2.java (hunk 1 above)
Comment (Manas-tarento) [reply src/main/java/org/sunbird/workflow/service/impl/WorkflowServiceImplV2.java:88]: moved this part into validate method

**Pull request closed without merging**